mixedbread_ai_haystack/embedders/text_embedder.py

```diff
@@ -1,145 +1,147 @@
-        :param instruction: Used to specify the instruction for the model. Can only be used with instruction based models like e5-large-v2
-        for example. If not specified, the default instruction for the model will be used.
-        """
+        super(MixedbreadAITextEmbedder, self).__init__(**kwargs)
 
-        self.model_name = model
+        self.model = model
         self.prefix = prefix
         self.suffix = suffix
         self.normalized = normalized
-        self.instruction = instruction
-        self._client = MixedbreadAi(
-            api_key=api_key,
-            base_url=base_url,
-            verify_ssl=verify_ssl,
-            timeout=timeout,
-            raise_for_status=True,
-            headers={
-                "User-Agent": "@mixedbread-ai/integrations-haystack",
-                **(custom_headers or {}),
-            }
-        )
+        self.encoding_format = encoding_format
+        self.truncation_strategy = truncation_strategy
+        self.dimensions = dimensions
+        self.prompt = prompt
 
-    def _get_telemetry_data(self) -> Dict[str, Any]:
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "MixedbreadAITextEmbedder":
         """
-        Data that is sent to Posthog for usage analytics.
+        Deserializes the component from a dictionary.
+
+        Parameters:
+            data (Dict[str, Any]): Dictionary to deserialize from.
+
+        Returns:
+            MixedbreadAITextEmbedder: The deserialized component.
         """
-        return {"model": self.model_name}
+        return from_dict(cls, data)
 
     def to_dict(self) -> Dict[str, Any]:
         """
-        This method overrides the default serializer in order to avoid leaking the `api_key` value passed
-        to the constructor.
+        Serializes the component to a dictionary.
+
+        Returns:
+            Dict[str, Any]: The serialized component data.
         """
+        parent_params = super(MixedbreadAITextEmbedder, self).to_dict()["init_parameters"]
+
         return default_to_dict(self,
-                               model=self.model_name,
+                               **parent_params,
+                               model=self.model,
                                prefix=self.prefix,
                                suffix=self.suffix,
                                normalized=self.normalized,
-                               instruction=self.instruction)
+                               encoding_format=self.encoding_format,
+                               truncation_strategy=self.truncation_strategy,
+                               dimensions=self.dimensions,
+                               prompt=self.prompt
+                               )
 
     @component.output_types(embedding=List[float], meta=Dict[str, Any])
-    def run(self, text: str):
-        """Embed a string."""
+    def run(self, text: str, prompt: Optional[str] = None) -> Dict[str, Any]:
```

Comment:
We could provide a more detailed type here, TypeDict f.e. 

Does @component... do this?

Reply:
a more detailed type for `meta`? should be possible

Reply:
added